test/new-e2e/tests/agent-subcommands/configcheck/configcheck_common_test.go

```diff
@@ -0,0 +1,114 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package configcheck
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type baseConfigCheckSuite struct {
+	e2e.BaseSuite[environments.Host]
+}
+
+type CheckConfigOutput struct {
+	CheckName  string
+	Filepath   string
+	InstanceID string
+	Settings   string
+}
+
+func MatchCheckToTemplate(checkname, input string) (*CheckConfigOutput, error) {
+	regexTemplate := fmt.Sprintf("=== %s check ===\n", checkname) +
+		"Configuration provider: file\n" +
+		"Configuration source: (?P<filepath>.*)\n" +
+		"Config for instance ID: (?P<instance>.*)\n" +
+		"(?P<settings>(?m:^[^=].*\n)+)" + // non-capturing group to get all settings
+		"==="
+	re := regexp.MustCompile(regexTemplate)
+	matches := re.FindStringSubmatch(input)
+
+	// without a match, SubexpIndex lookups panic with range errors
+	if len(matches) == 0 {
+		return nil, fmt.Errorf("regexp: no matches for %s check", checkname)
+	}
+
+	filepathIndex := re.SubexpIndex("filepath")
+	instanceIndex := re.SubexpIndex("instance")
+	settingsIndex := re.SubexpIndex("settings")
+
+	return &CheckConfigOutput{
+		CheckName:  checkname,
+		Filepath:   matches[filepathIndex],
+		InstanceID: matches[instanceIndex],
+		Settings:   matches[settingsIndex],
+	}, nil
+}
+
+func (v *baseConfigCheckSuite) TestMatchToTemplateHelper() {
```

Comment:
If I understand correctly this one is a unit test right? If yes I think it will never be executed anywhere, or within the e2e test job which can be confusing

Reply:
I changed it into a unit test. I believe its purpose is only to make sure the helper is working as expected